radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/ConstructionDeriveHandler.java

```diff
@@ -61,122 +61,75 @@
+	private final CoreModelMapper coreModelMapper;
 
 	@Inject
-	public ArchiveAccountHandler(
-		BerkeleyAccountInfoStore store,
-		BerkeleyAccountTxHistoryStore txHistoryStore,
-		BerkeleyTransactionsByIdStore txByIdStore,
-		Addressing addressing
-	) {
-		this.store = store;
-		this.txHistoryStore = txHistoryStore;
-		this.txByIdStore = txByIdStore;
-		this.addressing = addressing;
-	}
-
-	public JSONObject handleAccountGetBalances(JSONObject request) {
-		return withRequiredStringParameter(
-			request,
-			ADDRESS,
-			address -> addressing.forAccounts()
-				.parseFunctional(address)
-				.map(store::getAccountInfo)
-		);
-	}
-
-	public JSONObject handleAccountGetTransactionHistoryReverse(JSONObject request) {
-		return withRequiredParameters(
-			request,
-			List.of(ADDRESS, SIZE),
-			List.of(CURSOR, VERBOSE),
-			params -> allOf(
-				parseAddress(params),
-				safeLong(params, SIZE),
-				ok(params.optLong(CURSOR, -1)),
-				parseVerboseFlag(params)
-			).map((addr, limit, offset, verboseFlag) -> {
-					var txnArray = new JSONArray();
-					var lastOffset = new AtomicLong(0);
-					txHistoryStore.getTxnIdsAssociatedWithAccount(addr, offset < 0 ? null : offset)
-						.limit(limit)
-						.forEach(pair -> {
-							var json = txByIdStore.getTransactionJSON(pair.getFirst()).orElseThrow();
-							lastOffset.set(pair.getSecond());
-							txnArray.put(json);
-						});
-
-					var result = new JSONObject();
-					if (lastOffset.get() > 0) {
-						result.put(CURSOR, lastOffset.get() - 1);
-					}
-
-					return result
-						.put("totalCount", txnArray.length())
-						.put("transactions", txnArray);
-				})
-		);
-	}
-
-	public JSONObject handleAccountGetStakePositions(JSONObject request) {
-		return withRequiredStringParameter(
-			request,
-			ADDRESS,
-			address -> addressing.forAccounts().parseFunctional(address)
-				.map(store::getAccountStakes)
-				.map(JsonRpcUtil::wrapArray)
-		);
-	}
-
-	public JSONObject handleAccountGetUnstakePositions(JSONObject request) {
-		return withRequiredStringParameter(
-			request,
-			ADDRESS,
-			address -> addressing.forAccounts().parseFunctional(address)
-				.map(store::getAccountUnstakes)
-				.map(JsonRpcUtil::wrapArray)
-		);
-	}
+	ConstructionDeriveHandler(CoreModelMapper coreModelMapper) {
+		super(ConstructionDeriveRequest.class);
 
-	//-----------------------------------------------------------------------------------------------------
-	// internal processing
-	//-----------------------------------------------------------------------------------------------------
-	private Result<REAddr> parseAddress(JSONObject params) {
-		return addressing.forAccounts().parseFunctional(params.getString(ADDRESS));
+		this.coreModelMapper = coreModelMapper;
 	}
 
-	private Result<Boolean> parseVerboseFlag(JSONObject params) {
-		return ok(params.optBoolean(VERBOSE, false));
+	@Override
+	public ConstructionDeriveResponse handleRequest(ConstructionDeriveRequest request) throws CoreApiException {
```

Comment:
Is there a way to improve these long sequence of if/elses?

Reply:
Not with generated code. In the normal case this would be possible since polymorphism allows one to add abstracted logic to different types of objects. Not a possibility in this case because there's a clear separation between logic and data.

Reply:
I guess there is possibility of adding handlers into a `Map<Type, Handler>`